### DISCUSSION TITLE
As techempower benchmarks, will axum improve JSON and text performance

### DISCUSSION THREAD

**Discussion by joebnb** [tokio-rs/axum#2863] [General]
![image](https://github.com/user-attachments/assets/8aac9b73-1858-4c65-a010-6694f84c0c4e)

https://www.techempower.com/benchmarks/#section=data-r22&hw=ph&test=composite

**Comment by jplatte** [comment]
Last time I checked, the benchmark was still using axum 0.6. I did some improvements to it a while back, but ultimately it wasn't super interesting or gratifying work.

Even in performance-sensitive real world apps, I don't think it usually matters whether your framework is within 80% of the fastest, within 95% of it, or is the very fastest. I find it more important to have benchmarks within the codebase to ensure we don't have major regressions.

However, that doesn't mean that performance optimization work isn't welcome, it's just not a focus for me (and David, AFAIK). We just recently received a PR to reduce cloned [over here](https://github.com/tokio-rs/axum/pull/2865).

**Reply by andrew-james-dev** [reply]
Thank you for all your hard work on Axum; I've really enjoyed working with it. I've just created a PR to upgrade the benchmark to 0.7.5. Suggestions for improvement very much appreciated:
- https://github.com/TechEmpower/FrameworkBenchmarks/pull/9237
